pilot/pkg/bootstrap/server.go

```diff
@@ -1053,8 +1054,37 @@ func (s *Server) createPeerCertVerifier(tlsOptions TLSOptions) (*spiffe.PeerCert
 	return peerCertVerifier, nil
 }
 
+func checkPathsExist(paths ...string) bool {
+	for _, path := range paths {
+		fInfo, err := os.Stat(path)
+
+		if err != nil || fInfo.IsDir() {
+			return false
+		}
+	}
+	return true
+}
+
+// hasCustomTLSCerts returns the tls cert paths, used both if custom TLS certificates are configured via args or by mounting in well known.
+// while tls args should still take precedence the aim is to encourage loading the DNS tls cert in the well known path locations.
+func hasCustomTLSCerts(tlsOptions TLSOptions) (ok bool, tlsCertPath, tlsKeyPath, caCertPath string) {
+	// load from tls args as priority
+	if hasCustomTLSCertArgs(tlsOptions) {
+		return true, tlsOptions.CertFile, tlsOptions.KeyFile, tlsOptions.CaCertFile
+	}
+
+	if ok = checkPathsExist(constants.DefaultPilotTLSCert, constants.DefaultPilotTLSKey, constants.DefaultPilotTLSCaCert); ok {
```

Comment:
I am not sure the 'root CA' for custom DNS certs is required or commonly used. AFAIK just the key and cert are needed. 

What is the root used for ?

Reply:
Although it is not clear and I had to retrace the code path again, I think the root will be loaded / needed, when the certificatewatch is initiated it will load the CA into the trust bundle [here](https://github.com/istio/istio/blob/84d9f86021c64c65bc5cdc0b236da85d2f525907/pilot/pkg/bootstrap/certcontroller.go#L220) although a filewatcher is missing  (there is a seperate issue about that #37315).
At initialisation the above cacert will be retrieved when creating the (webhook patcher https://github.com/istio/istio/blob/b1bb7c1f2f413c8a1812b6c02187b03a4e729d03/pilot/pkg/bootstrap/sidecarinjector.go#L96) , which will be used to patch the webhook ca field
This does raise however the fact that the discovery server is still initiatised with args, so we need to ensure that's okay also: https://github.com/istio/istio/blob/84d9f86021c64c65bc5cdc0b236da85d2f525907/pilot/pkg/bootstrap/server.go#L735

To fully confirm this i will attempt to not set that flag as we were before and i am pretty certain there will manifest "unknown certificate authority errors" at least for any webhook calls to istiod.

The confusing thing is that when using istio-csr (citadel disabled), the root of trust management is handled by istio-csr which replicated the process in citadel that writes the `istio-root-ca-configmap` in every namespace.